### PR TITLE
feat: Introduce `@remix-run/eslint-config` package

### PR DESCRIPTION
**Pull request by chaance**
This PR creates a new package for providing simple ESLint support for new Remix projects. The goal is to be as unopinionated as possible while enabling rules that either deal with bugs or optimization. Like the rules for `create-react-app`, I also included an optional config for projects that use Jest.

In general, these rules are even less opinionated than the `react-app` config; we enable a subset of the rules they use and leave the rest up to users. Eventually I'd also like to create our own ESLint plugin that catches issues specific to Remix.

I have not updated `create-remix` to include this package/config by default, but would like to do that in a follow-up PR once this is merged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,21 +49,21 @@
...
     "eslint-import-resolver-typescript": "^2.5.0",
-    "eslint-plugin-flowtype": "^6.1.0",
-    "eslint-plugin-import": "^2.24.2",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-jest": "^25.2.2",
     "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-react": "^7.26.1",
-    "eslint-plugin-react-hooks": "^4.2.0",
+    "eslint-plugin-react-hooks": "alpha",
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply package.json:66]: [`v4.3.0`](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/CHANGELOG.md#430) is released with ESLint v8 compatibility

File: packages/create-remix/templates/_shared_js/app/entry.server.jsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ export default function handleRequest(
   responseHeaders,
   remixContext
 ) {
-  let markup = renderToString(
+  const markup = renderToString(
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/entry.server.jsx:10]: Note: Automatic fix via eslint --fix

File: packages/create-remix/templates/_shared_js/app/root.jsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ import globalStylesUrl from "~/styles/global.css";
 import darkStylesUrl from "~/styles/dark.css";
 
 // https://remix.run/api/app#links
-export let links = () => {
+export const links = () => {
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/root.jsx:16]: Note: Automatic fix via eslint --fix

File: packages/create-remix/templates/_shared_js/app/root.jsx [hunk 4]
```diff
@@ -58,7 +58,7 @@ export function ErrorBoundary({ error }) {
 
 // https://remix.run/api/conventions#catchboundary
 export function CatchBoundary() {
-  let caught = useCatch();
+  const caught = useCatch();
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/root.jsx:61]: Note: Automatic fix via eslint --fix

File: packages/create-remix/templates/_shared_js/app/routes/demos/about.jsx [hunk 5]
```diff
@@ -2,13 +2,13 @@ import { Outlet } from "remix";
 
 import stylesUrl from "~/styles/demos/about.css";
 
-export let meta = () => {
+export const meta = () => {
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/routes/demos/about.jsx:5]: Note: Automatic fix via eslint --fix

File: packages/create-remix/templates/_shared_js/app/routes/demos/actions.jsx [hunk 6]
```diff
@@ -8,9 +8,9 @@ export function meta() {
 // When your form sends a POST, the action is called on the server.
 // - https://remix.run/api/conventions#action
 // - https://remix.run/guides/data-updates
-export let action = async ({ request }) => {
-  let formData = await request.formData();
-  let answer = formData.get("answer");
+export const action = async ({ request }) => {
+  const formData = await request.formData();
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/routes/demos/actions.jsx:12]: Note: Automatic fix via eslint --fix

File: packages/create-remix/templates/_shared_js/app/routes/demos/params/$id.jsx [hunk 7]
```diff
@@ -3,7 +3,7 @@ import { json, useCatch, useLoaderData } from "remix";
 // The `$` in route filenames becomes a pattern that's parsed from the URL and
 // passed to your loaders so you can look up data.
 // - https://remix.run/api/conventions#loader-params
-export let loader = async ({ params }) => {
+export const loader = async ({ params }) => {
```
Comment (chaance) [inline packages/create-remix/templates/_shared_js/app/routes/demos/params/$id.jsx:6]: Note: Automatic fix via eslint --fix

**Pull request merged**